source/core/Math.cpp

```diff
@@ -350,7 +350,9 @@ DECLARE_SCALE2X_INTN_HELPER(Single)
          /* Args:  x loLimit hiLimit includeLo includeHi coercedOut  inRangeOut */\
         VIVM_TRACE_FUNCTION(InRangeAndCoerce##TYPE)    \
         if (_ParamPointer(5)) { \
-            if (_Param(1) > _Param(2)) { \
+            if (::isnan((double)_Param(0)) || ::isnan((double)_Param(1)) || ::isnan((double)_Param(2))) { \
```

Comment:
Is the "In Range" output argument affected if x or hiLimit or LoLimit is nan? Should we check for isnan there too? 

Would be worth adding checks for InRange out in the tests if not already done. 

Also, do we have tests for this node for both scalar and vector and Compare Elements and Compare Aggregates modes?

Reply:
_Param(6) should be set to False if any of the inputs are NaN, but it looks like the code already there that assigned _Param(6) would do this, because all the comparisons (>, ==) would return false.  And his tests verifies that.

Reply:
Also, should I even add these tests to their own test file? Or should I add them into InRangeAndCoerce.via as well?

Reply:
I meant coverage for scalars/vectors and Elements/Aggregates in general. If those tests are in the GVI test suite, that is sufficient. No need to add to .via file.